fog/test-client/src/test_client.rs

```diff
@@ -243,37 +255,40 @@ impl TestClient {
         client: &mut Client,
         transaction: &Tx,
     ) -> Result<BlockIndex, TestClientError> {
-        // Wait until ledger server can see all of these key images
-        let mut deadline = Some(Instant::now() + self.policy.tx_submit_deadline);
-        loop {
-            match client
-                .is_transaction_present(transaction)
-                .map_err(TestClientError::ConfirmTx)?
-            {
-                TransactionStatus::Appeared(block_index) => return Ok(block_index),
-                TransactionStatus::Expired => return Err(TestClientError::TxExpired),
-                TransactionStatus::Unknown => {}
-            }
-            deadline = if let Some(deadline) = deadline {
-                if Instant::now() > deadline {
-                    counters::TX_CONFIRMED_DEADLINE_EXCEEDED_COUNT.inc();
-                    if self.policy.fail_fast_on_deadline {
-                        return Err(TestClientError::SubmittedTxTimeout);
+        let tracer = tracer!();
```

Comment:
For my understanding, why are we making a new Tracer context here? There should be an outer one in test_transfer

Reply:
They all use the same global tracer context, it's easier than passing it around.